### DISCUSSION TITLE
Security for Ressources

### DISCUSSION THREAD

**Discussion by ZansDaus** [sourcebroker/t3api#76] [Q&A] [answered]
Hello,
i want to secure my API. Unfortunately the documentation for Security is missing.
Can someone show me some Examples or help me here?

Thanks!

**Comment by klamparski** [comment, marked as answer]
Hi @ZansDaus,

You can restrict access inside endpoint configuration, as on example below:

```
use SourceBroker\T3api\Annotation as T3api;

* @T3api\ApiResource(
*     collectionOperations={
*          "post"={
*              "path"="/comments",
*              "method"="POST",
*              "security"="frontend.user.isLoggedIn",
*          },
*     },
* )
class Comment extends \TYPO3\CMS\Extbase\DomainObject\AbstractEntity
{
}
 ```
 
 Inside `security` you can use all the [TYPO3's conditions using Symfony expression language](https://docs.typo3.org/m/typo3/reference-typoscript/main/en-us/Conditions/Index.html) and [write your custom ones](https://docs.typo3.org/m/typo3/reference-coreapi/10.4/en-us/ApiOverview/SymfonyExpressionLanguage/Index.html#sel-within-typoscript-conditions)
